### ISSUE TITLE
Doesn't build on x86_64

### ISSUE THREAD

**Issue by diversys** [HaikuArchives/Niue#3]
```

~/Niue> make -C Source  -i
make: Entering directory '/boot/home/Niue/Source'
Compiling BMapView.cpp...
BMapView.cpp:1307:1: error: prototype for 'long int BMapView::DrawThread()' does not match any in class 'BMapView'
 BMapView::DrawThread()
 ^
In file included from BMapView.cpp:28:0:
BMapView.h:58:20: error: candidate is: int32 BMapView::DrawThread()
    int32           DrawThread();
                    ^
Makefile:137: recipe for target 'obj.x86/BMapView.o' failed
make: [obj.x86/BMapView.o] Error 1 (ignored)
Compiling BSBar.cpp...
BSBar.cpp:29:1: error: prototype for 'BSBar::BSBar(BRect, const char*, port_id, long int, long int, orientation)' does not match any in class 'BSBar'
 BSBar::BSBar(BRect frame, const char *name,port_id drawport, long min, long max,
 ^
In file included from BSBar.cpp:26:0:
BSBar.h:29:7: error: candidates are: BSBar::BSBar(const BSBar&)
 class BSBar : public BScrollBar
       ^
BSBar.h:32:6: error:                 BSBar::BSBar(BRect, const char*, port_id, int32, int32, orientation)
      BSBar(BRect frame, const char *name, port_id drawport, int32 min, int32 max,
      ^
Makefile:137: recipe for target 'obj.x86/BSBar.o' failed
make: [obj.x86/BSBar.o] Error 1 (ignored)
Compiling NiueWindow.cpp...
NiueWindow.cpp:68:17: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
      {"\0\0",0} };
                 ^
NiueWindow.cpp:68:17: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:68:17: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:68:17: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:68:17: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:68:17: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:68:17: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:79:19: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
        {"\0\0",0} };
                   ^
NiueWindow.cpp:79:19: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:79:19: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:79:19: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:79:19: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:79:19: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:79:19: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:79:19: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:79:19: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:79:19: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:92:8: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
        };
        ^
NiueWindow.cpp:92:8: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:92:8: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:92:8: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:92:8: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:92:8: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:92:8: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:92:8: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:92:8: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:92:8: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:92:8: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:109:18: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
       {"\0\0",0} };
                  ^
NiueWindow.cpp:109:18: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:109:18: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:109:18: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:109:18: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:109:18: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:109:18: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:109:18: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:109:18: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:109:18: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:109:18: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:109:18: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:109:18: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:109:18: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:109:18: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:121:18: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
       {"\0\0",0} };
                  ^
NiueWindow.cpp:121:18: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:121:18: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:121:18: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:121:18: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:121:18: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:121:18: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:121:18: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:121:18: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:121:18: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:121:18: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:129:18: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
       {"\0\0",0} };
                  ^
NiueWindow.cpp:129:18: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:129:18: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:129:18: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:129:18: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:129:18: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:129:18: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:134:18: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
       {"\0\0",0} };
                  ^
NiueWindow.cpp:134:18: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:134:18: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:144:17: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
      {"\0\0",0} };
                 ^
NiueWindow.cpp:144:17: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:144:17: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:144:17: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:144:17: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:144:17: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:144:17: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:144:17: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp:144:17: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
NiueWindow.cpp: In constructor 'NiueWindow::NiueWindow(BRect, Buff*)':
NiueWindow.cpp:219:75: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
  fMapView = new BMapView(this,bb,rmyview,"NIUE:BMView",scbv,scbh,fDrawPort);
                                                                           ^
NiueWindow.cpp:221:45: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
  fListView = new listview(rlist, "fListView");
                                             ^
NiueWindow.cpp:223:49: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
  fOutputView = new OutputView(routput, "vwOuput");
                                                 ^
NiueWindow.cpp:237:54: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
  fStatusView = new StatusView(rssv,"StView",fDrawPort);
                                                      ^
NiueWindow.cpp:263:72: error: invalid conversion from 'long int (*)(void*)' to 'thread_func {aka int (*)(void*)}' [-fpermissive]
  fDrawThread = spawn_thread(MyDraw,tempname,B_DISPLAY_PRIORITY,fMapView);
                                                                        ^
In file included from /boot/system/develop/headers/os/interface/InterfaceDefs.h:10:0,
                 from /boot/system/develop/headers/os/interface/Menu.h:9,
                 from NiueWindow.cpp:22:
/boot/system/develop/headers/os/kernel/OS.h:331:18: note:   initializing argument 1 of 'thread_id spawn_thread(thread_func, const char*, int32, void*)'
 extern thread_id spawn_thread(thread_func, const char *name, int32 priority,
                  ^
NiueWindow.cpp: In destructor 'virtual NiueWindow::~NiueWindow()':
NiueWindow.cpp:291:36: error: cannot convert 'long int*' to 'status_t* {aka int*}' for argument '2' to 'status_t wait_for_thread(thread_id, status_t*)'
   wait_for_thread(fDrawThread,&temp);
                                    ^
NiueWindow.cpp: In member function 'virtual void NiueWindow::MessageReceived(BMessage*)':
NiueWindow.cpp:1012:18: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
     char* link = "http://www.tycomsystems.com/beos/BeBook/";    //then try online
                  ^
NiueWindow.cpp: In member function 'void NiueWindow::DoMenuFontName()':
NiueWindow.cpp:1569:58: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
   ProcTitle("%s Font",it,ifname,strcmp(curfam,myfam) == 0);
                                                          ^
NiueWindow.cpp: At global scope:
NiueWindow.cpp:1807:1: error: prototype for 'STextView::STextView(BRect, const char*, uint32, int32)' does not match any in class 'STextView'
 STextView::STextView(BRect frame, const char *name, uint32 resizeMask, int32 msg)
 ^
In file included from NiueWindow.cpp:28:0:
NiueWindow.h:159:7: error: candidates are: STextView::STextView(const STextView&)
 class STextView:public BTextView
       ^
NiueWindow.h:162:5: error:                 STextView::STextView(BRect, const char*, uint32, long int)
     STextView(BRect frame,const char *name,uint32 flags,long msg);
     ^
Makefile:137: recipe for target 'obj.x86/NiueWindow.o' failed
make: [obj.x86/NiueWindow.o] Error 1 (ignored)
Compiling UBuffer.cpp...
UBuffer.cpp:288:7: error: prototype for 'int32 Buff::AddCharReal(long int, const char*, long int)' does not match any in class 'Buff'
 int32 Buff::AddCharReal(long cn,const char *s,long cnt)
       ^
In file included from UBuffer.cpp:53:0:
UBuffer.h:105:10: error: candidate is: int32 Buff::AddCharReal(int32, const char*, int32)
    int32 AddCharReal(int32 cnum,const char *source,int32 length);
          ^
UBuffer.cpp: In member function 'status_t Buff::MyExec(NiueWindow*, int32, char*)':
UBuffer.cpp:738:10: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
   args[0]="/bin/sh";
          ^
UBuffer.cpp:739:10: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
   args[1]="-c";
          ^
UBuffer.cpp: In member function 'int32 Buff::NextError(NiueWindow*, char*, int32*)':
UBuffer.cpp:868:30: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
    if (mstrcmp(nu,"File \"",6)) break;
                              ^
UBuffer.cpp:871:29: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
    if (mstrcmp(nu,"; Line",6)) break;
                             ^
UBuffer.cpp: In function 'char* MyMimeType(int)':
UBuffer.cpp:965:12: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
   case 0:tt="";break;
            ^
UBuffer.cpp:966:12: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
   case 1:tt="text/x-source-code";break;
            ^
UBuffer.cpp:967:12: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
   case 2:tt="text/plain";break;
            ^
UBuffer.cpp:968:12: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
   case 3:tt="text/html";break;
            ^
UBuffer.cpp:969:12: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
   case 4:tt="text/x-email";break;
            ^
UBuffer.cpp:970:12: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
   case 5:tt="text/x-source-code";break;
            ^
UBuffer.cpp: In member function 'int32 Buff::Paste(BMessage*)':
UBuffer.cpp:980:76: error: no matching function for call to 'BMessage::FindData(const char [11], <anonymous enum>, const void**, int32*)'
  status_t rr=msg->FindData("text/plain",B_MIME_TYPE,(const void**)&data,&ll);
                                                                            ^
In file included from /boot/system/develop/headers/os/support/Archivable.h:10:0,
                 from /boot/system/develop/headers/os/app/Handler.h:12,
                 from /boot/system/develop/headers/os/app/Looper.h:13,
                 from /boot/system/develop/headers/os/support/Autolock.h:10,
                 from UBuffer.cpp:30:
/boot/system/develop/headers/os/app/Message.h:269:15: note: candidate: status_t BMessage::FindData(const char*, type_code, const void**, ssize_t*) const
    status_t   FindData(const char* name, type_code type,
               ^
/boot/system/develop/headers/os/app/Message.h:269:15: note:   no known conversion for argument 4 from 'int32* {aka int*}' to 'ssize_t* {aka long int*}'
/boot/system/develop/headers/os/app/Message.h:271:15: note: candidate: status_t BMessage::FindData(const char*, type_code, int32, const void**, ssize_t*) const
    status_t   FindData(const char* name, type_code type,
               ^
/boot/system/develop/headers/os/app/Message.h:271:15: note:   candidate expects 5 arguments, 4 provided
UBuffer.cpp:987:68: error: no matching function for call to 'BMessage::FindData(const char [11], <anonymous enum>, const void**, int32*)'
   rr=msg->FindData("text/plain",B_MIME_TYPE,(const void**)&data,&ll);
                                                                    ^
In file included from /boot/system/develop/headers/os/support/Archivable.h:10:0,
                 from /boot/system/develop/headers/os/app/Handler.h:12,
                 from /boot/system/develop/headers/os/app/Looper.h:13,
                 from /boot/system/develop/headers/os/support/Autolock.h:10,
                 from UBuffer.cpp:30:
/boot/system/develop/headers/os/app/Message.h:269:15: note: candidate: status_t BMessage::FindData(const char*, type_code, const void**, ssize_t*) const
    status_t   FindData(const char* name, type_code type,
               ^
/boot/system/develop/headers/os/app/Message.h:269:15: note:   no known conversion for argument 4 from 'int32* {aka int*}' to 'ssize_t* {aka long int*}'
/boot/system/develop/headers/os/app/Message.h:271:15: note: candidate: status_t BMessage::FindData(const char*, type_code, int32, const void**, ssize_t*) const
    status_t   FindData(const char* name, type_code type,
               ^
/boot/system/develop/headers/os/app/Message.h:271:15: note:   candidate expects 5 arguments, 4 provided
UBuffer.cpp: In member function 'void Buff::CCMsg(int32, int32, NiueWindow*)':
UBuffer.cpp:1062:8: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
    str = "Cut";
        ^
UBuffer.cpp:1068:8: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
    str = "Copy";
        ^
UBuffer.cpp:1073:8: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
    str = "Remove";
        ^
UBuffer.cpp: In member function 'void Buff::MessageReceived2(BMessage*)':
UBuffer.cpp:1178:8: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
    text="\n";
        ^
Makefile:137: recipe for target 'obj.x86/UBuffer.o' failed
make: [obj.x86/UBuffer.o] Error 1 (ignored)
updating: ../Skeleton/ (stored 0%)
updating: ../Skeleton/Makefile (deflated 58%)
updating: ../Skeleton/yourapp.cpp (deflated 39%)
updating: ../Skeleton/yourapp.h (deflated 38%)
updating: ../Skeleton/yourapp.rsrc (deflated 87%)
updating: ../Skeleton/yourappview.cpp (deflated 44%)
updating: ../Skeleton/yourappview.h (deflated 29%)
updating: ../Skeleton/yourappwindow.cpp (deflated 39%)
updating: ../Skeleton/yourappwindow.h (deflated 38%)
updating: 1. About Niue (deflated 44%)
updating: 2. Project setup (deflated 52%)
updating: 3. Code completion (deflated 36%)
updating: 4. Exporting (deflated 44%)
updating: 5. Snippets (deflated 41%)
updating: 6. Visual Designer (deflated 45%)
updating: 7. Known bugs (deflated 33%)
updating: 8. Shortcuts (deflated 50%)
Linking...
cc: error: obj.x86/BMapView.o: No such file or directory
cc: error: obj.x86/BSBar.o: No such file or directory
cc: error: obj.x86/NiueWindow.o: No such file or directory
cc: error: obj.x86/UBuffer.o: No such file or directory
Makefile:128: recipe for target 'Niue' failed
make: [Niue] Error 1 (ignored)
...xres and mimeset...
Niue executable created successfully
make: Leaving directory '/boot/home/Niue/Source'
~/Niue> 
```

**Comment by waddlesplash** [comment]
The source is a massive mess that I have basically no interest in fixing tbh.

**Comment by diversys** [comment]
Fixed in 290a4f0c1b99ea960db2d3ebc75bfcdc8f015783.
